crates/linker/src/macho.rs

```diff
@@ -601,453 +630,292 @@ fn gen_macho_le(
     // https://github.com/kubkon/zignature
     // https://github.com/kubkon/zig-deploy
 
-    use macho::{Section64, SegmentCommand64};
+    use macho::{
+        Section64, SegmentCommand64, VM_PROT_EXECUTE as E, VM_PROT_READ as R, VM_PROT_WRITE as W,
+    };
 
     let exec_header = load_struct_inplace::<macho::MachHeader64<LE>>(exec_data, 0);
+    let size_of_header = mem::size_of::<macho::MachHeader64<LE>>();
     let num_load_cmds = exec_header.ncmds.get(LE);
     let size_of_cmds = exec_header.sizeofcmds.get(LE) as usize;
 
-    // Add a new text segment and data segment
+    // Add a new text segment and data segment load commands.
+    // We assume existence of __ROC_TEXT,__text, __ROC_DATA_CONST,__const, __ROC_DATA,__data and __ROC_DATA,__bss.
```

Comment:
Aren't `__ROC_*` segments the ones we are adding? So they wouldn't exist yet?

Reply:
Correct. All we do here is insert load commands pointing at the offset/vmaddr of 0 and with 0 size. Even if unpopulated later it will not hinder the loader so we can either leave it in or prune the unused ones in the surgical linking step. I've added it here since we most likely will need to create most if not all of them in the linking step, but it could very well be done in that step too. Anyhow, think of it as a placeholder for the real thing which the link step will actually populate with real data.